central/securedclustercertgen/certificates_test.go

```diff
@@ -182,3 +229,60 @@ func (s *localScannerSuite) TestServiceIssueLocalScannerCerts() {
 	}
 	testutils.MustUpdateFeature(s.T(), features.ScannerV4, scannerV4Enabled)
 }
+
+// TestServiceIssueSecuredClusterCerts checks the issuance of certificates for secured clusters.
+func (s *securedClusterCertgenSuite) TestServiceIssueSecuredClusterCerts() {
+	getServiceTypes := func() set.FrozenSet[string] {
```

Comment:
nit:

Given that what you are actually extracting here are the *serviceTypeNames*, how about adjusting the name of this var here?

While we are at it, I think we can simplify this a bit.

Instead of defining a new function here, which allocates new slices, iterates over them, calls append multiple times, creates a new frozen set, how about we actually execute this function *once* (instead of once per loop iteration) and save the result as some var `allServiceTypeNames`.

And line 277, which is currently

```
expectedServiceTypes := getServiceTypes().Unfreeze()
```

could then be changed to create a dumb copy:
```
expectedServiceTypeNames := append([]string{}, allServiceNameTypes)
```

Reply:
I removed the unneeded `serviceTypes` variable, but I think a set is more appropriate here than a slice.
We remove found certificates from the set, and expect it to be empty at the end. Changing to a []string slice will make that logic more complicated.